app/src/main/java/ch/epfl/sdp/appart/MapActivity.java

```diff
@@ -30,33 +29,67 @@
     @Inject
     LocationService locationService;
 
+    @Inject
+    MapService mapService;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_map);
 
+        PermissionRequest.askForLocationPermission(this, () -> {
+            Log.d("PERMISSION", "Location permission granted");
+            setupMap();
+        }, () -> {
+            Log.d("PERMISSION", "Refused");
+            finish();
+        }, () -> Log.d("PERMISSION", "Popup"));
+
+    }
+
+
+    /*
+        To implement the feature :
+            - Set map center on user location
+            - Get location from all the ads (or all the ads in the country),
+            see if it is possible to get ad only
+                if they satisfy a condition.
+                It would be great if I could get all the ads (location only
+                because if we have millions of ad this is going to be huge,
+                or restrict by country for instance)
+                , then transform the address into latitude and longitude,
+                then maybe it is possible to ask the map object if a
+                   specific location is on the map. If it is display it.
```

Comment:
It is! you can query firestore with a `where` method to filter results that contain a certain field with a certain value. Maybe in that case we would need to modify the content of the ads (for example to include the country), I see it more difficult to filter by radius though, int hat case we would have to retrieve the whole result and filter locally. We could for example get partial ads where we keep locally ad id and location, filter locally and retrieve the full ads that satisfy the filtering on location

Reply:
Ok nice, I will try to make that work in the PR that will improve the map.